### PR TITLE
Feature/article meta

### PR DESCRIPTION
**Pull request by ghost**
Changing article meta tag model to not use composite keys as Eloquent does not support them.

### CODE REVIEW

File: app/src/app/admin/articles/article/article.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ namespace app.admin.articles.article {
                         return articleService.getArticle($stateParams.permalink)
                             .then((article) => {
 
-                                article._articleMetas = articleService.hydrateMetaFromTemplate(article._articleMetas, ArticleConfig.articleMetaTemplate);
+                                article._articleMetas = articleService.hydrateMetaFromTemplate(article, ArticleConfig.articleMetaTemplate);
```
Comment (zakhenry) [inline app/src/app/admin/articles/article/article.ts]: Why the change to passing in the full article object? It then requires that the function is aware of the structure of the article. I think the way you had it is better with pasing the articleMetas collection. Maybe it would make more sense to rename the function `hydrateMetaCollectionFromTemplate` ?

File: api/app/Models/ArticleMeta.php [hunk 2]
```diff
@@ -21,9 +21,7 @@ class ArticleMeta extends BaseModel
      */
     public $table = 'article_metas';
 
-    protected $primaryKey = 'meta_name';
-
-    protected $fillable = ['meta_name', 'meta_content'];
+    protected $fillable = ['id', 'article_id', 'meta_name', 'meta_content'];
```
Comment (zakhenry) [inline api/app/Models/ArticleMeta.php]: The primary key for all entities must match the route segment + `_id` so in this case it would be `meta_id`. See the other models for the pattern.

This is for clarity of what id it is when writing sql, for the purposes of views etc.

File: app/src/app/admin/articles/article/article.spec.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@ namespace app.admin.articles.article {
 
         let seededChance = new Chance(1),
             testMeta:common.models.ArticleMeta = new common.models.ArticleMeta({
+                articleId: undefined,
+                id: 'f6a31e1d-60e0-35b5-a878-085deabe57bd',
```
Comment (zakhenry) [inline app/src/app/admin/articles/article/article.spec.ts]: use `seededChance.guid()` to create a mock uuid fixture.

File: app/src/app/admin/articles/article/article.ts (hunk 1 above)
Comment (ghost) [inline app/src/app/admin/articles/article/article.ts]: The function requires the article ID to hydrate the meta tags so I figured it would be better to pass the whole article rather than have an additional parameter to call the function with.

Comment (zakhenry) [inline app/src/app/admin/articles/article/article.ts]: Ah ok I see the rationale, but yea I think it would be better to have the article id passed as a param.
